### ISSUE TITLE
useEffect executes many times during render when using options

### ISSUE THREAD

**Issue by pocesar** [trekhleb/use-position#1]
when using the second parameter, the `useEffect` gets called many times, leading to multiple calls to onChange, and multiple location changes, even if they aren't happening. either put on the readme that you need to pass an `useMemo`'ed options object or useMemo inside the hook

**Comment by kcoyner** [comment]
@pocesar  Can you post an example of how you implemented useMemo in this hook?   Thank you.

**Comment by pocesar** [comment]
@kcoyner 

```jsx
const settings = useMemo(() => {
  return { enableHighAccuracy: true }
}, [])

const { latitude, longitude, timestamp, accuracy, error } = usePosition(true, settings)
```

**Comment by pumppi** [comment]
I'm having same issue.

**Comment by samhwang** [comment]
Hi @trekhleb 

Can you please have at this issue and also the fix provided by @mewben at #8 . This problem has been sitting here for 6 months, and its been super annoying because it triggers the re-render of the component like an infinite loop. :(

**Comment by hally9k** [comment]
@trekhleb - Should we be using useSubscription here? See: https://github.com/facebook/react/pull/15022/files#diff-f3a2eff7618256a34a255026486e7e9fR7

**Comment by trekhleb** [comment]
Hey folks, it should be fixed now in `0.0.7`. Thanks @mewben for the fix!

**Comment by techieeliot** [comment]
@trekhleb I'm having this issue crop up in `0.0.7`. 

I'm fetching the geolocation for my weather app data, and it's repeatedly calling the api to fetch the weather.
![Screen Shot 2020-05-13 at 4 43 05 PM](https://user-images.githubusercontent.com/19453294/81869004-e4f21900-9538-11ea-86d1-a8e99353da41.png)

# Thanks for creating this great tool. How would I reduce the number of times that I'm calling to the API?
